ckanext/hierarchy/helpers.py

```diff
@@ -73,19 +76,58 @@ def traverse_highlight(group_tree, name_list):
         traverse_highlight(group, selected_names)
     return group_tree_list
 
+
 def get_allowable_parent_groups(group_id):
+
+    # Get config
+    PARENT_GROUP_TYPE = config.get('hierarchy.parent_group_type', 'organization')
+    PARENT_GROUP_SHOULD_BELONG_TO_USER = p.toolkit.asbool(
+        config.get('hierarchy.parent_group_should_belong_to_user', False))
+
+    # Get groups
     if group_id:
         group = model.Group.get(group_id)
         allowable_parent_groups = \
-            group.groups_allowed_to_be_its_parent(type='organization')
+            group.groups_allowed_to_be_its_parent(type=PARENT_GROUP_TYPE)
     else:
         allowable_parent_groups = model.Group.all(
-            group_type='organization')
+            group_type=PARENT_GROUP_TYPE)
+
+    # Filter by user
+    if PARENT_GROUP_SHOULD_BELONG_TO_USER:
+        user_orgs = p.toolkit.get_action('organization_list_for_user')({}, {})
+        def predicat(group):
+            for user_org in user_orgs:
+                if user_org['id'] == group.id and user_org['capacity'] == 'admin':
```

Comment:
I think if you belong to an organization in any role (eg `editor` or `member`) you should see the parent groups, so drop the `and user_org['capacity'] == 'admin'` check.